### PR TITLE
Fix/macos vllm

### LINKED ISSUE #148: scandeval tries to install vllm on Apple silicon
vllm does not currently seem to install on Apple silicon (M2 Max). Maybe it should not be installed for this platform? If that's the solution, I could make a PR.

(scandeval) petersk@ADM-133336-Mac ScandEval % pip install .
Processing /Users/petersk/Nobackup/ScandEval
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: accelerate<1.0.0,>=0.23.0 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.23.0)
Requirement already satisfied: bert-score<1.0.0,>=0.3.13 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.3.13)
Requirement already satisfied: click<9.0.0,>=8.1.3 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (8.1.7)
Requirement already satisfied: datasets<3.0.0,>=2.15.0 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (2.16.1)
Requirement already satisfied: demjson3<4.0.0,>=3.0.6 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (3.0.6)
Requirement already satisfied: einops<1.0.0,>=0.6.0 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.7.0)
Requirement already satisfied: evaluate<1.0.0,>=0.4.1 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.4.1)
Requirement already satisfied: flax<1.0.0,>=0.6.3 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.7.4)
Requirement already satisfied: huggingface-hub<1.0.0,>=0.7.0 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.20.2)
Requirement already satisfied: jax<1.0.0,>=0.4.1 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.4.17)
Requirement already satisfied: jaxlib<1.0.0,>=0.4.1 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.4.17)
Requirement already satisfied: levenshtein<1.0.0,>=0.21.1 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.22.0)
Requirement already satisfied: numpy<2.0.0,>=1.23.0 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (1.26.0)
Requirement already satisfied: openai<1.0.0,>=0.27.8 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.28.1)
Requirement already satisfied: pandas<2.0.0,>=1.4.0 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (1.5.3)
Requirement already satisfied: protobuf<3.21.0,>=3.20.0 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (3.20.3)
Requirement already satisfied: pyinfer<1.0.0,>=0.0.3 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.0.3)
Requirement already satisfied: python-dotenv<1.0.0,>=0.20.0 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.21.1)
Requirement already satisfied: requests<3.0.0,>=2.31.0 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (2.31.0)
Requirement already satisfied: rouge-score<1.0.0,>=0.1.2 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.1.2)
Requirement already satisfied: sacremoses<1.0.0,>=0.0.53 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.0.53)
Requirement already satisfied: sentencepiece<1.0.0,>=0.1.96 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.1.99)
Requirement already satisfied: seqeval<2.0.0,>=1.2.2 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (1.2.2)
Requirement already satisfied: termcolor<3.0.0,>=2.0.0 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (2.3.0)
Requirement already satisfied: tiktoken<1.0.0,>=0.4.0 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (0.5.1)
Requirement already satisfied: torch<3.0.0,>=2.1.1 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (2.1.2)
Requirement already satisfied: tqdm<5.0.0,>=4.62.0 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (4.66.1)
Requirement already satisfied: transformers<5.0.0,>=4.36.2 in /Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages (from ScandEval==9.1.1) (4.36.2)
Collecting vllm<1.0.0,>=0.2.7 (from ScandEval==9.1.1)
  Using cached vllm-0.2.7.tar.gz (170 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [23 lines of output]
      /private/var/folders/cm/27_svh8s6px54vddfs6sjp4m0000gq/T/pip-build-env-bq3590db/overlay/lib/python3.10/site-packages/torch/nn/modules/transformer.py:20: UserWarning: Failed to initialize NumPy: No module named 'numpy' (Triggered internally at /Users/runner/work/pytorch/pytorch/pytorch/torch/csrc/utils/tensor_numpy.cpp:84.)
        device: torch.device = torch.device(torch._C._get_default_device()),  # torch.device('cpu'),
      Traceback (most recent call last):
        File "/Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/Users/petersk/anaconda3/envs/scandeval/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/private/var/folders/cm/27_svh8s6px54vddfs6sjp4m0000gq/T/pip-build-env-bq3590db/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/private/var/folders/cm/27_svh8s6px54vddfs6sjp4m0000gq/T/pip-build-env-bq3590db/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/cm/27_svh8s6px54vddfs6sjp4m0000gq/T/pip-build-env-bq3590db/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 230, in <module>
        File "/private/var/folders/cm/27_svh8s6px54vddfs6sjp4m0000gq/T/pip-build-env-bq3590db/overlay/lib/python3.10/site-packages/torch/utils/cpp_extension.py", line 1076, in CUDAExtension
          library_dirs += library_paths(cuda=True)
        File "/private/var/folders/cm/27_svh8s6px54vddfs6sjp4m0000gq/T/pip-build-env-bq3590db/overlay/lib/python3.10/site-packages/torch/utils/cpp_extension.py", line 1203, in library_paths
          if (not os.path.exists(_join_cuda_home(lib_dir)) and
        File "/private/var/folders/cm/27_svh8s6px54vddfs6sjp4m0000gq/T/pip-build-env-bq3590db/overlay/lib/python3.10/site-packages/torch/utils/cpp_extension.py", line 2416, in _join_cuda_home
          raise OSError('CUDA_HOME environment variable is not set. '
      OSError: CUDA_HOME environment variable is not set. Please set it to your CUDA install root.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

### PR DESCRIPTION
**Pull request by saattrupdan**
### Fixed
- There was a bug where all models were removed from disk prior to benchmarking. This will now only happen if the `--clear-model-cache` flag is set.
- The `vllm` package cannot be installed when CUDA is not available - this is now neither installed nor used when this is the case, and generative few-shot evaluation is done using the `transformers` package rather than `vllm`.

This closes #148

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -18,6 +18,9 @@ jobs:
         matrix:
             os: [ubuntu-latest]
             python-version: ["3.10", "3.11"]
+            include:
+              - os: macos-latest
+                python-version: "3.10"
```
Comment (KennethEnevoldsen) [inline .github/workflows/ci.yaml:23]: Oh didn't know that was a option

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (saattrupdan) [reply .github/workflows/ci.yaml:23]: Same! I just learnt about it myself 🙂

Comment (KennethEnevoldsen) [reply .github/workflows/ci.yaml:23]: Will def. use it somewhere in the future thanks!

**Pull request merged**